From 311f93c636128bd0ee7cb944c843c58f71e0e0f7 Mon Sep 17 00:00:00 2001
From: Ketan Patil <ketanp@nvidia.com>
Date: Wed, 1 Jul 2026 07:46:25 +0000
Subject: [PATCH] video: tegra: nvmap: Fix null pointer deref issue

nvmap_handle pointer was being dereferenced without checking for NULL
value, fix it by adding a check.

Bug 6355805

Change-Id: Ie3bd4e0915c08d62693a91936a75880061739aea
Signed-off-by: Ketan Patil <ketanp@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-nv-oot/+/3655915
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Pritesh Raithatha <praithatha@nvidia.com>
---
 drivers/video/tegra/nvmap/nvmap_ioctl.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/video/tegra/nvmap/nvmap_ioctl.c b/drivers/video/tegra/nvmap/nvmap_ioctl.c
index 0bd10bed..18dab01f 100644
--- a/drivers/video/tegra/nvmap/nvmap_ioctl.c
+++ b/drivers/video/tegra/nvmap/nvmap_ioctl.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-only
 /*
- * Copyright (c) 2011-2024, NVIDIA CORPORATION. All rights reserved.
+ * Copyright (c) 2011-2026, NVIDIA CORPORATION. All rights reserved.
  *
  * User-space interface to nvmap
  */
@@ -1636,15 +1636,17 @@ int nvmap_ioctl_get_fd_from_list(struct file *filp, void __user *arg)
 
 	for (i = 0; i < op.num_handles; i++) {
 		hs[i] = nvmap_handle_get_from_id(client, hndls[i]);
-		tot_hs_size += hs[i]->size;
 		if (IS_ERR_OR_NULL(hs[i])) {
-			pr_err("invalid handle_ptr[%d] = %u\n",
-				i, hndls[i]);
-			while (i--)
-				nvmap_handle_put(hs[i]);
+			pr_err("Invalid input handle id: %u\n", hndls[i]);
+
+			/* Clean up handles (index i-1 to 0) */
+			while (i != 0)
+				nvmap_handle_put(hs[--i]);
+
 			err = -EINVAL;
 			goto free_mem;
 		}
+		tot_hs_size += hs[i]->size;
 	}
 
 	/* Add check for sizes of all the handles should be > offs and size */
-- 
GitLab

